cpp/examples/build.sh

```diff
@@ -25,6 +25,12 @@ build_example() {
   cmake -S ${example_dir} -B ${build_dir} -Dcudf_ROOT="${LIB_BUILD_DIR}"
   # Build
   cmake --build ${build_dir} -j${PARALLEL_LEVEL}
+
+  # Install only if CMAKE_INSTALL_PREFIX is set
+  if [[ -n "${CMAKE_INSTALL_PREFIX}" ]]; then
+    echo "Setting prefix = ${CMAKE_INSTALL_PREFIX}"
+    cmake --install ${build_dir} --prefix /home/coder/testing/basic
```

Comment:
I think you're supposed to set `CMAKE_INSTALL_PREFIX` (the environment variable in our case, since we want it to be controlled by `install_cudf_examples.sh` rather than this script) when CMake runs its configure step, then you shouldn't need the `--prefix` here. I'm still learning this part of CMake so I'd be happy to have corrections.

Reply:
Still cleaning these up but the `--prefix` option in `cmake --install` is pretty much useless and does not take any precedence over the initially set `CMAKE_INSTALL_PREFIX`.

Reply:
The install rules in this example are written in a way that is fundamentally incompatible with `cmake --install --prefix`.

The cmake install command docs outline the issue:
```
DESTINATION <dir>
Specify the directory on disk to which a file will be installed. <dir> should be a relative path. An absolute path is allowed, but not recommended.

When a relative path is given it is interpreted relative to the value of the [CMAKE_INSTALL_PREFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html#variable:CMAKE_INSTALL_PREFIX) variable. The prefix can be relocated at install time using the DESTDIR mechanism explained in the [CMAKE_INSTALL_PREFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html#variable:CMAKE_INSTALL_PREFIX) variable documentation.

As absolute paths do not work with the cmake --install command's [--prefix](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake--install-0) option, or with the [cpack](https://cmake.org/cmake/help/latest/manual/cpack.1.html#manual:cpack(1)) installer generators, it is strongly recommended to use relative paths throughout for best support by package maintainers. In particular, there is no need to make paths absolute by prepending [CMAKE_INSTALL_PREFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html#variable:CMAKE_INSTALL_PREFIX); this prefix is used by default if the DESTINATION is a relative path. 
```

In all the install rules you are using `${CMAKE_INSTALL_PREFIX}` which is not needed since all relative paths are evaluatd based off of it, and secondly creating absolute paths which can't be re-rooted by `--prefix`.